api/nodes/serializers.py

```diff
@@ -124,8 +131,20 @@ def update(self, instance, validated_data):
         the request to be in the serializer context.
         """
         assert isinstance(instance, Node), 'instance must be a Node'
+        auth = self.get_user_auth(self.context['request'])
         for attr, value in validated_data.items():
-            setattr(instance, attr, value)
+            if attr == 'tags':
+                old_tags = set([tag._id for tag in instance.tags])
+                if value:
+                    current_tags = set(value)
+                else:
+                    current_tags = set()
+                for new_tag in (current_tags - old_tags):
+                    instance.add_tag(new_tag, auth=auth)
+                for deleted_tag in (old_tags - current_tags):
```

Comment:
What if the client wants to remove all of a node's tags? If I understand correctly, this isn't possible with this logic.

What is the best practice for updating list fields in a REST API?